File: app/src/main/java/in/nisb/nisbapp/NotificationFragment.java
package in.nisb.nisbapp;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by mridul on 28/3/17.
 */

public class NotificationFragment extends Fragment {
    View view;
    SwipeRefreshLayout sr;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.activity_notifs,container,false);

        loadList();
        TextView tv_clear = (TextView) view.findViewById(R.id.notifs_clear);
        tv_clear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                NisbUser.clearNotifications(getContext());
                loadList();
            }
        });


        sr = (SwipeRefreshLayout) view.findViewById(R.id.notifs_refresh);
        sr.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                loadList();
            }
        });

        return view;
    }
    void loadList(){
        ListView lv = (ListView) view.findViewById(R.id.notifs_list);
        JSONObject notifs_list[] = NisbUser.getNotifications(getContext());
        if (notifs_list!=null)
            lv.setAdapter(new NotificationAdapter(getContext(),notifs_list));
        else {
            String empty[] = {};
            lv.setAdapter(new ArrayAdapter<String>(getContext(),R.layout.notifs_list_item,empty));
        }


        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                TextView tv_data = (TextView) view.findViewById(R.id.notifs_data);
                try {
                    JSONObject jo = new JSONObject(tv_data.getText().toString());
                    String context = jo.getString("context");
                    Intent intent;
                    if (context.equals("blog-single")){
                        intent = new Intent(getContext(),BlogSingleActivity.class);
                        intent.putExtra("title",jo.getString("title"));
                        intent.putExtra("extra",jo.getString("extra"));
                        intent.putExtra("content",jo.getString("content"));
                        startActivity(intent);
                    }
                    if (context.equals("event-single")){
                        intent = new Intent(getContext(),EventSingleActivity.class);
                        intent.putExtra("id",jo.getString("id"));
                        startActivity(intent);
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
        ((SwipeRefreshLayout) view.findViewById(R.id.notifs_refresh)).setRefreshing(false);

    }

    private class NotificationAdapter extends ArrayAdapter<JSONObject>{
        Context c;
        JSONObject jo[];
        public NotificationAdapter(Context con,JSONObject j[]){
            super(con,R.layout.notifs_list_item,j);
            c =con;
            jo = j;
        }
        public View getView(int position, View convertView, ViewGroup parent) {
            convertView = LayoutInflater.from(c).inflate(R.layout.notifs_list_item, parent, false);
            TextView tv_title = (TextView) convertView.findViewById(R.id.notifs_text);
            TextView tv_data = (TextView) convertView.findViewById(R.id.notifs_data);
            try {
                String title = jo[position].getString("body");
                tv_title.setText(title);
                tv_data.setText(jo[position].toString());
            }catch (JSONException j){}

            return convertView;
        }
    }
}

File: app/src/main/java/in/nisb/nisbapp/fragments/NotificationFragment.kt
package `in`.nisb.nisbapp

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.widget.SwipeRefreshLayout
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView

import org.json.JSONException
import org.json.JSONObject

/**
 * Created by mridul on 28/3/17.
 */

class NotificationFragment : Fragment() {
    internal var view: View? = null
    internal var sr: SwipeRefreshLayout? = null
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        view = inflater!!.inflate(R.layout.activity_notifs, container, false)

        loadList()
        val tv_clear = view?.findViewById(R.id.notifs_clear) as TextView
        tv_clear.setOnClickListener {
            DatabaseFunctions().clearNotifications(context)
            loadList()
        }


        sr = view?.findViewById(R.id.notifs_refresh) as SwipeRefreshLayout
        sr?.setOnRefreshListener { loadList() }

        return view
    }

    internal fun loadList() {
        val lv = view?.findViewById(R.id.notifs_list) as ListView
        val notifs_list = DatabaseFunctions().getNotifications(context)
        if (notifs_list != null)
            lv.adapter = NotificationAdapter(context, notifs_list)
        else {
            val empty = arrayOf<String>()
            lv.adapter = ArrayAdapter(context, R.layout.notifs_list_item, empty)
        }


        lv.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val tv_data = view?.findViewById(R.id.notifs_data) as TextView
            try {
                val jo = JSONObject(tv_data.text.toString())
                val context = jo.getString("context")
                var intent: Intent
                if (context == "blog-single") {
                    intent = Intent(getContext(), BlogSingleActivity::class.java)
                    intent.putExtra("url", jo.getString("url"))
                    startActivity(intent)
                }
                if (context == "event-single") {
                    intent = Intent(getContext(), EventSingleActivity::class.java)
                    intent.putExtra("id", jo.getString("id"))
                    startActivity(intent)
                }
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        }
        (view?.findViewById(R.id.notifs_refresh) as SwipeRefreshLayout).isRefreshing = false

    }

    private inner class NotificationAdapter(internal var c: Context, internal var jo: Array<JSONObject?>?) : ArrayAdapter<JSONObject>(c, R.layout.notifs_list_item, jo) {
        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {

            var cV = convertView ?: LayoutInflater.from(c).inflate(R.layout.notifs_list_item, parent, false)

            val tv_title = cV.findViewById(R.id.notifs_text) as TextView
            val tv_data = cV.findViewById(R.id.notifs_data) as TextView
            try {
                val title = jo!![position]!!.getString("body")
                tv_title.text = title
                tv_data.text = jo!![position].toString()
            } catch (j: JSONException) {
            }

            return cV
        }
    }
}
